### PR TITLE
Use evented dataclass for scalebar

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Continues from #1910 by using an eventdata class for our scalebar. #1910 should be reviewed first, as this PR contains changes from that one too.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by sofroniewn** [comment]
Ok, this is ready to review. I've merged in the changes from #1910 and now you can see how beautifully simple this transition to evented dataclass is!!!

File: napari/components/scale_bar.py [hunk 1]
```diff
@@ -1,108 +1,38 @@
...
+from ..utils.events.dataclass import Property, dataclass
 from ._viewer_constants import Position
 
 
+@dataclass(events=True, properties=True)
 class ScaleBar:
     """Scale bar indicating size in world coordinates.
 
-    Attributes
+    Parameters
```
Comment (jni) [inline napari/components/scale_bar.py]: Not sure about this change...?

**Review by jni — APPROVED** [review]
Wow. =D

File: napari/components/scale_bar.py (hunk 1 above)
Comment (sofroniewn) [inline napari/components/scale_bar.py]:
```suggestion
    Attributes
```

Comment (sofroniewn) [reply napari/components/scale_bar.py]: Reverted!

**Pull request merged**